crates/matrix-sdk/src/room/edit.rs

```diff
@@ -234,6 +252,101 @@ async fn make_edit_event<S: EventSource>(
     }
 }
 
+/// Sets the caption of a media event content.
+///
+/// Why a macro over a plain function: the event content types all differ from
+/// each other, and it would require adding a trait and implementing it for all
+/// event types instead of having this simple macro.
+macro_rules! set_caption {
+    ($event:expr, $caption:expr) => {
+        let filename = $event.filename().to_owned();
+        // As a reminder:
+        // - body and no filename set means the body is the filename
+        // - body and filename set means the body is the caption, and filename is the
+        //   filename.
+        if let Some(caption) = $caption {
+            $event.filename = Some(filename);
+            $event.body = caption;
+        } else {
+            $event.filename = None;
+            $event.body = filename;
+        }
+    };
+}
+
+/// Sets the caption of a [`RoomMessageEventContent`].
+///
+/// Returns true if the event represented a media event (and thus the captions
```

Comment:
Again, why a boolean and not a proper error?

Reply:
I'm not sure I follow, how can the user of a `update_media_caption` know from the signature that the only possible error is an `not media event` one? From the outside a number of things might've gone wrong. Why go throug mental gymanstics based on *only the current implementation* when you can make it explicit through an error (that can also grow in time as need be)?

Reply:
I wouldn't call reading the doc comment "mental gymnastics", to be fair :-) I'm not using an error type because [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) — and if we need to distinguish among several error kinds later, we can add the error when we need to. I find that using a `Result` here would be optimizing for future possibilities that don't exist quite yet, or, if you prefer, a tiny act of over-engineering.

Reply:
As you wish then 👍